### PR TITLE
add a variable to control whether prefer pyls

### PR DESCRIPTION
**Pull request by kevindragon**
https://github.com/seagle0128/.emacs.d/issues/156

### CODE REVIEW

**Comment by seagle0128** [comment]
The proriority is able to select the lsp server. Is there any reason to add a variable for control it? I don't think it's necessary.

**Comment by kevindragon** [comment]
I did not know how to set priority. If the lsp-python-ms is the recommended client, it should be the default, in my opinion.

**Comment by seagle0128** [comment]
Currently there are two lsp servers for Python: pyls and mspyls. `pyls` is builtin of `lsp-mode` and it has higher priority. `mspyls` is implemented by C# and it's faster, but it's not the default one. 

If `pyls` is not installed, lsp-mode will try to use `mspyls` if `lsp-pyton-ms` is loaded.
This behavior is consistent as other servers. For example, gopls, bingo, go-langserver for GO, ts-ls, js-ls for JS, etc.

**Comment by kevindragon** [comment]
如果我希望使用`pyls`就没有必要安装`lsp-python-ms`这个包了。既然我安装了`lsp-python-ms`，但是默认又不启动，这个行为我觉得不是太好。我之前安装的201912XX版本是默认启动的，我升级后花了两个小时来检查哪里出了问题（可能是我对lsp-mode以及lsp-python-ms的源代码不熟悉）。
所以我觉得既然安装了`lsp-python-ms`这个包，就应该默认开启，如果不需要，uninstall就可以了。

File: lsp-python-ms.el [hunk 1]
```diff
@@ -41,6 +41,12 @@
 ;; Forward declare variable
 (defvar lsp-render-markdown-markup-content)
 
+(defvar lsp-python-ms-prefer-pyls
```
Comment (seagle0128) [inline lsp-python-ms.el:44]: Should use `defcustom`.

File: README.org [hunk 2]
```diff
@@ -12,11 +13,16 @@ A minimal ~use-package~ initialization might be:
...
   (use-package lsp-python-ms
     :ensure t
+    ;; :init (setq lsp-python-ms-prefer-pyls nil)
     :hook (python-mode . (lambda ()
                             (require 'lsp-python-ms)
                             (lsp))))  ; or lsp-deferred
 #+END_SRC
 
+#+BEGIN_QUOTE
+For now, the pyls is the default lsp client (priority of lsp-python-ms is set to -2), If you prefer lsp-python-ms, please set =lsp-python-ms-prefer-pyls= to =nil=.
```
Comment (seagle0128) [inline README.org:23]: or uninstall `pyls`?

**Review by seagle0128 — CHANGES_REQUESTED** [review]

**Comment by seagle0128** [comment]
Add a new option `lsp-python-ms-auto-install-server` to handle this case.

**Pull request closed without merging**